core/modules/image/image.field.inc

```diff
@@ -759,32 +824,74 @@ function image_field_formatter_view($entity_type, $entity, $field, $instance, $l
   }
 
   foreach ($items as $delta => $item) {
-    // Add class for floating the image
-    if (!empty($display['settings']['image_float'])) {
-      $item['attributes']['class'][] = 'align-' . $display['settings']['image_float'];
-    }
+    if (!empty($item)) {
+      // Add class for floating the image.
+      if (!empty($display['settings']['image_float'])) {
+        $item['attributes']['class'][] = 'align-' . $display['settings']['image_float'];
+      }
+
+      if (!empty($display['settings']['image_load'])) {
+        // Although available in Chromium, the 'auto' value is not mentioned in
+        // the specification. Since it may be subject to change, we recommend
+        // not to use it until it gets officially included.
+        if ($display['settings']['image_load'] != 'auto') {
+          $item['attributes']['loading'] = $display['settings']['image_load'];
+        }
+      }
 
-    if (!empty($display['settings']['image_load'])) {
-      // Although available in Chromium, the 'auto' value is not mentioned in
-      // the specification. Since it may be subject to change, we recommend not
-      // to use it until it gets officially included.
-      if ($display['settings']['image_load'] != 'auto') {
-        $item['attributes']['loading'] = $display['settings']['image_load'];
+      if (isset($link_file)) {
+        $uri = array(
+          'path' => file_create_url($item['uri']),
+          'options' => array(),
+        );
+      }
+
+      $is_svg = image_is_svg($item['uri']);
+
+      // Add custom attributes only for SVG images.
+      if ($is_svg) {
+        $image_svg_dimensions = $display['settings']['image_svg_dimensions'];
+
+        foreach ($image_svg_dimensions as $attribute => $attribute_value) {
+          if (!empty($attribute)) {
+            $item[$attribute] = $attribute_value;
+          }
+        }
       }
     }
 
-    if (isset($link_file)) {
-      $uri = array(
-        'path' => file_create_url($item['uri']),
-        'options' => array(),
+    if (!$is_svg || $display['settings']['image_svg_display']) {
+      $element[$delta] = array(
+        '#theme' => 'image_formatter',
+        '#item' => $item,
+        '#image_style' => $display['settings']['image_style'],
+        '#path' => isset($uri) ? $uri : '',
       );
     }
-    $element[$delta] = array(
-      '#theme' => 'image_formatter',
-      '#item' => $item,
-      '#image_style' => $display['settings']['image_style'],
-      '#path' => isset($uri) ? $uri : '',
-    );
+    else {
+      $svg_path = file_create_url($item['uri']);
+      $svg = file_get_contents($svg_path);
+      $svg_clean = filter_xss($svg, array('svg', 'use', 'title', 'desc', 'defs', 'linearGradient', 'stop', 'rect', 'circle', 'path'));
```

Comment:
Oops, this does a little too much. It breaks perfectly harmless and valid svg's. :stuck_out_tongue_winking_eye: 

Maybe after some discussion we'll end up to always just display SVG in the image :point_left: formatter as img tag, and leave plain SVG as markup to contrib... :shrug:

Reply:
Oh, it's only sanitizing on inline display. If that's the case, only having an inline img tag option is ok by me.

Reply:
The latest commit seems to have fixed the sanitizer - it no longer breaks a valid SVG. :+1: 

Whether we allow direct markup display in core, or leave it to contrib, is something still to determine, I guess.